tests/cli/bootsys/test_mgmt_power.py

```diff
@@ -177,3 +183,230 @@ def test_mgmt_ncns_skip_prompt_power_off(self):
+            mock_logger.info.assert_any_call('Successfully set next boot device to disk (Boot0001) for ncn-w003')
+
+    def test_set_next_boot_device_to_disk_command_fail(self):
+        """Test that the function handles command execution failures and continues"""
+        # Mock successful connection for all NCNs
+        self.mock_ssh_client.connect.side_effect = [None, None, None]
+
+        # Mock command execution failure for the first NCN and success for the rest
+        mock_stdout_1 = MagicMock()
+        mock_stdout_2 = MagicMock()
+        mock_stderr = MagicMock()
+        mock_stdout_1.channel.recv_exit_status.return_value = 1
+        mock_stdout_1.read.return_value = b""
+        mock_stdout_2.channel.recv_exit_status.return_value = 0
+        mock_stdout_2.read.return_value = b"Boot0001* UEFI OS\nBoot0002* Other OS"
+
+        self.mock_ssh_client.exec_command.side_effect = [
+            (None, mock_stdout_1, mock_stderr),
+            (None, mock_stdout_2, mock_stderr),
+            (None, mock_stdout_2, mock_stderr)
+        ]
+
+        with patch('sat.cli.bootsys.mgmt_power.LOGGER') as mock_logger:
+            ncns = ['ncn-w001', 'ncn-w002', 'ncn-w003']
+            set_next_boot_device_to_disk(self.mock_ssh_client, ncns)
+
+            expected_connect_calls = [call('ncn-w001'), call('ncn-w002'), call('ncn-w003')]
+            self.mock_ssh_client.connect.assert_has_calls(expected_connect_calls)
+
+            expected_exec_calls = [
+                call('efibootmgr'),
+                call('efibootmgr -n 0001'),
+                call('efibootmgr'),
+                call('efibootmgr -n 0001')
+            ]
+            mock_ssh_client.exec_command.assert_has_calls(expected_exec_calls, any_order=True)
+
+            mock_logger.warning.assert_called_with(
+                'Unable to determine boot order of ncn-w001, efibootmgr exited with exit code 1')
+            mock_logger.info.assert_any_call('Successfully set next boot device to disk (Boot0001) for ncn-w002')
+            mock_logger.info.assert_any_call('Successfully set next boot device to disk (Boot0001) for ncn-w003')
+
+    def test_set_next_boot_device_to_disk_command_fail(self):
+        """Test that the function handles command execution failures"""
+        mock_ssh_client = MagicMock()
+        mock_stdout = MagicMock()
+        mock_stderr = MagicMock()
+        mock_stdout.channel.recv_exit_status.side_effect = [1, 0]
+        mock_stdout.read.return_value = b""
+
+        mock_ssh_client.exec_command.return_value = (None, mock_stdout, mock_stderr)
+
+        with patch('sat.cli.bootsys.mgmt_power.LOGGER') as mock_logger:
+            ncns = ['ncn-w001']
+            set_next_boot_device_to_disk(mock_ssh_client, ncns)
+
+            mock_logger.warning.assert_called_with(
+                'Unable to determine boot order of ncn-w001, efibootmgr exited with exit code 1'
+            )
```

Comment:
Here's my suggested test that tests when just one `efibootmgr -n` command fails for a set of three NCNs:

```suggestion
    def test_set_next_boot_device_to_disk_set_command_fails(self):
        """Test that the function handles a failure when setting the next boot device for a node"""
        mock_ssh_client = MagicMock()
        mock_stdout_list = MagicMock()
        mock_stdout_set = MagicMock()
        mock_stderr = MagicMock()

        # Simulate successful listing of boot entries
        mock_stdout_list.channel.recv_exit_status.return_value = 0
        mock_stdout_list.read.side_effect = [
            b"Boot0001* UEFI OS\nBoot0002* Other OS",  # Output for ncn-w001
            b"Boot0003* UEFI OS\nBoot0004* Other OS",  # Output for ncn-w002
            b"Boot0005* UEFI OS\nBoot0006* Other OS"   # Output for ncn-w003
        ]

        # Simulate failure when setting the next boot device for ncn-w002
        mock_stdout_set.channel.recv_exit_status.side_effect = [0, 1, 0]
        mock_stdout_set.read.return_value = b"BootNext: 0001"

        mock_ssh_client.exec_command.side_effect = [
            (None, mock_stdout_list, mock_stderr),  # Call to efibootmgr for ncn-w001
            (None, mock_stdout_set, mock_stderr),   # Call to efibootmgr -n 0001 for ncn-w001
            (None, mock_stdout_list, mock_stderr),  # Call to efibootmgr for ncn-w002
            (None, mock_stdout_set, mock_stderr),   # Call to efibootmgr -n 0001 for ncn-w002
            (None, mock_stdout_list, mock_stderr),  # Call to efibootmgr for ncn-w003
            (None, mock_stdout_set, mock_stderr)    # Call to efibootmgr -n 0001 for ncn-w003
        ]

        with patch('sat.cli.bootsys.mgmt_power.LOGGER') as mock_logger:
            ncns = ['ncn-w001', 'ncn-w002', 'ncn-w003']
            set_next_boot_device_to_disk(mock_ssh_client, ncns)

            expected_exec_calls = [
                call('efibootmgr'),
                call('efibootmgr -n 0001'),
                call('efibootmgr'),
                call('efibootmgr -n 0003'),
                call('efibootmgr'),
                call('efibootmgr -n 0005')
            ]
            mock_ssh_client.exec_command.assert_has_calls(expected_exec_calls)

            mock_logger.info.assert_any_call('Successfully set next boot device to disk (Boot0001) for ncn-w001')
            mock_logger.warning.assert_called_with(
                'Failed to set next boot device for ncn-w002, efibootmgr -n 0003 exited with exit code 1'
            )
            mock_logger.info.assert_any_call('Successfully set next boot device to disk (Boot0005) for ncn-w003')

```

Reply:
Just as a note, this unit test was generated with GitHub Copilot and then modified by me with further assistance from GitHub Copilot as well as some manual changes.

As a result, it mostly follows the style of the other tests here. For example, it patches the `LOGGER` object instead of using `assertLogs`. One thing that I thought was interesting (and nice!) is that it gave good descriptive variable names to my two Mock objects for mocking stdout. It named them `mock_stdout_list` and `mock_stdout_set` as opposed to just numbering them.

Reply:
Looks very nice and interesting, will add these changes.